pytradfri/color.py

```diff
@@ -0,0 +1,48 @@
+from .const import (ATTR_LIGHT_COLOR_X as X, ATTR_LIGHT_COLOR_Y as Y)
+
+
+KNOWN_XY = {
+  2200: {X: 33135, Y: 27211},
+  2700: {X: 30140, Y: 26909},
+  4000: {X: 24930, Y: 24694},
+}
+
+KNOWN_KELVIN = KNOWN_XY.keys()
+
+KNOWN_X = [v[X] for v in KNOWN_XY.values()]
+
+
+def can_kelvin_to_xy(k):
+    return k is not None and min(KNOWN_KELVIN) <= k <= max(KNOWN_KELVIN)
+
+
+def can_x_to_kelvin(x):
+    return x is not None and min(KNOWN_X) <= x <= max(KNOWN_X)
+
+
+def kelvin_to_xy(k):
+    if k in KNOWN_XY:
```

Comment:
It's not caching - this constant stores known X and Y params for provided color temperatures. Then It uses this constant to estimate X and Y for unknown kelvin values between known values.